ddtrace/internal/utils/cache.py

```diff
@@ -38,23 +39,27 @@ def get(self, key, f):  # type: ignore[override]
         function ``f`` is called on the key to generate it. The return value is
         then stored in the cache and returned to the caller.
         """
-        if len(self) >= self.maxsize:
-            for _, h in zip(range(self.maxsize >> 1), sorted(self, key=lambda h: self[h][1])):
-                del self[h]
 
         _ = super(LFUCache, self).get(key, miss)
         if _ is not miss:
-            value, count = _
-            self[key] = (value, count + 1)
+            with self.count_lock:
```

Comment:
Should we also acquire a lock for the `get(key, miss)`  call?

Reply:
the operation is atomic, so we should be good here